Commonalities/documentation/Working/user-consent.md

```diff
@@ -0,0 +1,396 @@
+2. During delegated authorized process (jwt-bearer flow), the **auth server** can validate as pre-requisite to issue an access token if there is user consent (opt-in) registered for the user and the requested purpose/s. 
+
+Operator NaaS platform MUST define **Consent API** and **delegated authorization** (jwt-bearer flow) to meet current requirements in telco operators, where user consents are gathered and managed by a third-party element and not by Auth server itself.
+
+The API allows an application back-end to perform the following actions on behalf of a specific user:
+
+- Find if it has the consent to fulfill a certain purpose.
+- Create a consent
+- Revoke a consent
+- Simply inform that the consent was prompted to the user but he did not accept it
+
+But there is another API required, referred to as the **GDPR API**. This API is required for the management and provision of purposes, legal entities, and PI scopes. In order to request access to Personal Information under certain purpose, that purpose must be previously provisioned (see example in [Purposes](#purposes) section), and the PI scopes associated properly defined. In the same way, user consent is created for the legal entity associated to the client application and those legal entities must be also provisioned.
+
+So typically, the steps that an application backend will need to follow having an access token (let's call it _access\_token\_1_) for a logged in user (login authorization) would be as follows:
+
+1. Try to get a second access token (let's call it _access\_token\_2_) for the non-automatic purposes associated to the application.
+2. If an error is received or if in the `purpose` field of the token response there are one or more missing purposes for which the application needs to be granted access to, then application will need to ask the user for consent.
+3. Using _access\_token\_1_, get the information from the GDPR API to prompt the user for consent for each of the purposes the application requires (i.e. "opt-in" field of the provisioned purpose).
+4. Using _access\_token\_1_, create the consents captured through the Consent API.
+5. Wait for the completion of the consent creation (status polling by `operation_id`)
+6. Repeat step 1 and now the application back-end should receive all the purposes the application needs listed in the `purpose` field of the token response you get. This is _access\_token\_2_.
+7. Use _access\_token\_2_ to consume the scopes the application requires.
+
+Below you can see a flow diagram explaining these steps for an application that wants to consume the _Health API_ under the purpose _determine-health_.
+
+![Consent gathering with Consent API flow](./images/userconsent-2.png)
```

Comment:
It would be very helpful to replace this picture with the picture sent by jesus.penagarcia-oliva@telefonica.com as it is more elaborate

Reply:
Ok, we will look into it since we are planning to commit a doc update with changes considering the feedback received so far from partners and operators and considering new scenarios around NaaS like IoT ones.

Reply:
Thanks for the info. Would it be also possible to add the feedback you have received from Camara members under some  issue/pull request (as comment may be). This could be interesting for others as well and would also stop people from asking duplicate questions.

Reply:
Is there also an intention to extend AuthN&AuthZ once this has been accepted as a deliverable?

Reply:
I'm waiting to have permission to push changes to WorkingGroups repository :)
I have already a local branch ready to be pushed to create a PR for AuthN/AuthZ commonality.